Async init shouldn't be required inside async actor

```py3
---------------------------------------------------------------------------
RayTaskError(AssertionError)              Traceback (most recent call last)
<ipython-input-17-3d607351d5d7> in <module>
----> 1 ray.get(a.something.remote())

~/Desktop/ray/ray/python/ray/worker.py in get(object_ids, timeout)
   1480                     worker.core_worker.dump_object_store_memory_usage()
   1481                 if isinstance(value, RayTaskError):
-> 1482                     raise value.as_instanceof_cause()
   1483                 else:
   1484                     raise value

RayTaskError(AssertionError): ray::AsyncActor.something() (pid=26320, ip=10.142.32.76)
  File "python/ray/_raylet.pyx", line 643, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 621, in function_executor
  File "/Users/simonmo/miniconda3/lib/python3.6/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/Users/simonmo/miniconda3/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "<ipython-input-6-b3532e96dec8>", line 7, in something
  File "python/ray/includes/unique_ids.pxi", line 199, in ray._raylet.ObjectID.__await__
  File "/Users/simonmo/Desktop/ray/ray/python/ray/async_compat.py", line 38, in get_async
    async_api_init()
  File "/Users/simonmo/Desktop/ray/ray/python/ray/experimental/async_api.py", line 98, in init
    "You are using async_api inside a running event loop. "
AssertionError: You are using async_api inside a running event loop. Please call `await _async_init()` to initialize inside asynchrounous context.


In [20]: @ray.remote
    ...: class AsyncActor:
    ...:     async def something(self):
    ...:         import asyncio
    ...:         await _async_init()
    ...:         await asyncio.sleep(0.)
    ...:         print("Done")
    ...:         result = await task.remote() <- somethow this requires asyncio API to be initialized
    ...:         print(result)
    ...:         ray.get(task.remote())
```

--- a/python/ray/_raylet.pyx
+++ b/python/ray/_raylet.pyx
@@ -1173,8 +1173,8 @@ cdef class CoreWorker:
         if self.async_event_loop is None:
             self.async_event_loop = asyncio.new_event_loop()
             asyncio.set_event_loop(self.async_event_loop)
-            # Initialize the async plasma connection in worker
-            # Delayed import due to everything depends on raylet
+            # Initialize the async plasma connection.
+            # Delayed import due to async_api depends on _raylet.
             from ray.experimental.async_api import _async_init
             self.async_event_loop.run_until_complete(_async_init())
         if self.async_thread is None:

--- a/python/ray/_raylet.pyx
+++ b/python/ray/_raylet.pyx
@@ -1173,6 +1173,10 @@ cdef class CoreWorker:
         if self.async_event_loop is None:
             self.async_event_loop = asyncio.new_event_loop()
             asyncio.set_event_loop(self.async_event_loop)
+            # Initialize the async plasma connection in worker
+            # Delayed import due to everything depends on raylet
+            from ray.experimental.async_api import _async_init
+            self.async_event_loop.run_until_complete(_async_init())
         if self.async_thread is None:
             self.async_thread = threading.Thread(
                 target=lambda: self.async_event_loop.run_forever()

--- a/python/ray/tests/py3_test.py
+++ b/python/ray/tests/py3_test.py
@@ -259,5 +259,5 @@ def test_asyncio_actor_async_get(ray_start_regular_shared):
         async def get(self):
             return await remote_task.remote()
 
-    getter = AsyncGetter.options(is_direct_call=True, is_asyncio=True).remote()
+    getter = AsyncGetter.options(is_asyncio=True).remote()
     assert ray.get(getter.get.remote()) == 1

--- a/python/ray/tests/py3_test.py
+++ b/python/ray/tests/py3_test.py
@@ -247,3 +247,17 @@ async def test_asyncio_get(ray_start_regular_shared, event_loop):
 
     with pytest.raises(ray.exceptions.RayTaskError):
         await ray.async_compat.get_async(direct.throw_error.remote())
+
+
+def test_asyncio_actor_async_get(ray_start_regular_shared):
+    @ray.remote
+    def remote_task():
+        return 1
+
+    @ray.remote
+    class AsyncGetter:
+        async def get(self):
+            return await remote_task.remote()
+
+    getter = AsyncGetter.options(is_direct_call=True, is_asyncio=True).remote()
+    assert ray.get(getter.get.remote()) == 1

